pulp_puppet_plugins/pulp_puppet/plugins/importers/metadata.py

```diff
@@ -42,12 +42,9 @@ class InvalidTarball(ExtractionException):
 CHECKSUM_READ_BUFFER_SIZE = 65536
 
 
-def extract_metadata(filename, temp_dir, module=None):
+def extract_metadata(filename, temp_dir):
     """
-    Pulls the module's metadata file out of the module's tarball and updates the
-    module instance with its contents. The module instance itself is updated
-    as part of this call. It is up to the caller to delete the temp_dir after
-    this executes.
+    Pulls the module's metadata file out of the module's tarball and returns it.
```

Comment:
Consider replacing with the simpler sentence: `Returns the module's metadata from the tarball`